### PR TITLE
chore: extract guardian logic in its own package

### PR DESCRIPTION
**Pull request by bluecco**
### Issue / feature description

Move guardian logic in its own package

### Changes

Moved all the logic and files used by guardian in `guardian` package

### Checklist

- [x] Rebased to the last commit of the target branch (or merged)
- [x] Code self-reviewed
- [x] Code self-tested
- [x] Tests updated (if needed)
- [x] All tests are passing locally

Please keep your pull request as small as possible. If you need to make multiple changes, please create separate pull requests for each change. Create a draft pull request if you need early feedback. This will mark the pull request as a work in progress and prevent it from being reviewed or merged until you are ready.

Please move drafts to the ready for review (regular PR) when you are ready to start the review process and other developers will pick it up from there.

### CODE REVIEW

**Comment by bluecco** [comment]
@janek26 @simonheys this is how I've proceeded in order to have guardian in its own package

@simonheys maybe we can check together how to test that everything is ok

File: packages/guardian/package.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{
+  "name": "@argent/guardian",
+  "version": "6.0.1",
```
Comment (simonheys) [inline packages/guardian/package.json:3]:
```suggestion
  "version": "5.2.0",
```

File: packages/extension/package.json [hunk 2]
```diff
@@ -80,6 +80,7 @@
     "@argent/x-multicall": "^5.2.0",
     "@argent/x-sessions": "^5.2.0",
     "@argent/x-swap": "^5.2.0",
+    "@argent/guardian": "^6.0.1",
```
Comment (simonheys) [inline packages/extension/package.json:83]:
```suggestion
    "@argent/guardian": "^5.2.0",
```

File: packages/guardian/.gitignore [hunk 3]
```diff
@@ -0,0 +1,24 @@
+# Logs
```
Comment (simonheys) [inline packages/guardian/.gitignore:1]: This whole file can be removed unless its specifying something other than what's in the root `.gitignore`?

Comment (bluecco) [reply packages/guardian/.gitignore:1]: useless, will remove

File: packages/guardian/package.json (hunk 1 above)
Comment (bluecco) [reply packages/guardian/package.json:3]: right, I was starting from web wallet at the beginning

File: packages/guardian/package.json [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  "devDependencies": {
+    "@tsconfig/svelte": "^3.0.0",
+    "@types/async-retry": "^1.4.5",
+    "starknet": "^4.19.2",
+    "typescript": "^4.9.3",
+    "vite": "^3.0.0",
+    "vite-plugin-dts": "^1.4.0"
+  },
+  "peerDependencies": {
+    "starknet": "^4.19.2"
```
Comment (simonheys) [inline packages/guardian/package.json:39]: This is specified in both dev and peer -  should be one or the other?

Comment (bluecco) [reply packages/guardian/package.json:39]: will remove get-starknet-core, not used

File: packages/guardian/src/services/account.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
+import { getJwt } from "./jwt"
+
+export const ARGENT_API_BASE_URL = process.env.ARGENT_API_BASE_URL as string
```
Comment (simonheys) [inline packages/guardian/src/services/account.ts:3]: I feel that `ARGENT_API_BASE_URL` and `cosignerSign` implementations are specific to the extension. e.g. the original fetcher method of cosignerSign also includes some version headers that are expected.

How about moving the types `Cosigner, CosignerMessage` into `GuardianSigner.ts` where they are referenced, then move the actual `cosignerSign` implementation back into the extension?

Comment (bluecco) [reply packages/guardian/src/services/account.ts:3]: works for me, I can always change that later if needed to be shared

File: packages/guardian/src/services/idb.ts [hunk 6]
```diff
@@ -1,17 +1,26 @@
-import Dexie, { Table } from "dexie"
```
Comment (janek26) [inline packages/guardian/src/services/idb.ts:1]: I'm not sure if we should keep this in shared, or if we should move everything storage out of this shared package, and just export a StorageInterface ?

Comment (simonheys) [reply packages/guardian/src/services/idb.ts:1]: I agree to move it out of shared, the use case seems slightly different.

Comment (bluecco) [reply packages/guardian/src/services/idb.ts:1]: 👍

File: packages/guardian/src/services/account.ts (hunk 5 above)
Comment (bluecco) [reply packages/guardian/src/services/account.ts:3]: also @simonheys , for ARGENT_API_BASE_URL, I'll add the url in `cosignerSign`

resulting into `export const cosignerSign: Cosigner = async (url: string, message: CosignerMessage) => {...}`

File: packages/guardian/src/services/idb.ts (hunk 6 above)
Comment (simonheys) [reply packages/guardian/src/services/idb.ts]: Works well!

File: packages/guardian/package.json [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    "build": "vite build",
+    "dev": "vite build --watch",
+    "prepare": "yarn build"
+  },
+  "dependencies": {
+    "jose": "^4.11.2",
+    "lodash-es": "^4.17.21"
+  },
+  "devDependencies": {
+    "@tsconfig/svelte": "^3.0.0",
```
Comment (simonheys) [inline packages/guardian/package.json:29]: This package is not using Svelte? Could simplify by removing this and maybe using same tsconfig from similar package - maybe multicall package?

**Review by simonheys — CHANGES_REQUESTED** [review]
Works great - just one final comment on config then good to merge

**Review by simonheys — APPROVED** [review]

**Pull request merged**